app/controllers/spree/admin/errors_controller.rb

```diff
@@ -1,11 +1,17 @@
 module Spree
   module Admin
     class ErrorsController < BaseController
+
       skip_before_action :authorize_admin
 
       def forbidden
+        authorize! :read, ::Spree::AdminPanel
         render status: 403
       end
+
+      rescue_from CanCan::AccessDenied do |_exception|
```

Comment:
What do we need this one for? I think this will also change the behavior of other controllers, right?

Reply:
In `Spree::Core::ControllerHelper::Auth` there is access handling (which I guess fires when we are denied access to a reserce):
```ruby
rescue_from CanCan::AccessDenied do |_exception|
    redirect_unauthorized_access
end
```
which created theoretically infinite loop of redirecting to /admin/forbidden, being denied access, redirecting to /admin/forbidden... Which resulted in error:
![Screenshot 2024-01-10 at 10 20 17](https://github.com/spree/spree_backend/assets/26871147/0975fec9-d681-466e-a5be-10fb180787bd)

I decided that plain 404 error page is more clear, so I overwritten said `rescue_from CanCan::AccessDenied`